api_app/services/health_checker.py

```diff
@@ -1,24 +1,59 @@
+from contextlib import asynccontextmanager
+
 from azure.core import exceptions
 from azure.cosmos import CosmosClient
+from azure.identity.aio import DefaultAzureCredential
+from azure.servicebus.aio import ServiceBusClient
+from azure.servicebus.exceptions import ServiceBusConnectionError
+from fastapi import HTTPException, status
 
 from api.dependencies.database import get_store_key
 from core import config
 from models.schemas.status import StatusEnum
 from resources import strings
 
 
+@asynccontextmanager
+async def default_credentials():
+    """
+    Context manager which yields the default credentials.
+    """
+    credential = DefaultAzureCredential(managed_identity_client_id=config.MANAGED_IDENTITY_CLIENT_ID)
```

Comment:
Will this work when we debug this locally? (Running not as a MSI?)

Reply:
Seems like it works. I debugged locally